File: app/src/main/java/com/stackanswer/retrofit/ApiService.java
package com.stackanswer.retrofit;

import com.stackanswer.retrofit.detailshow.Response;
import com.stackanswer.retrofit.listMoviePopuler.ListMoviePopularResponse;
import com.stackanswer.retrofit.listShowPopular.ListShowPopularResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @GET("3/movie/popular?")
    Call<ListMoviePopularResponse> getListMoviePopuler(
            @Query("api_key") String api_key,
            @Query("language") String language,
            @Query("page") String page
    );

    @GET("3/tv/popular?")
    Call<ListShowPopularResponse> getListShowPopuler(
            @Query("api_key") String api_key,
            @Query("language") String language,
            @Query("page") String page
    );

    @GET("3/tv/{id_movie}?")
    Call<Response> getDetailShowPopuler(
            @Path("id_movie") String id_movie,
            @Query("api_key") String api_key,
            @Query("language") String language
    );

    @GET("3/movie/{id_movie}?")
    Call<com.stackanswer.retrofit.detailmovie.Response> getDetailMoviePopuler(
            @Path("id_movie") String id_movie,
            @Query("api_key") String api_key,
            @Query("language") String language
    );
}

File: app/src/main/java/com/stackanswer/source/network/ApiService.kt
package com.stackanswer.source.network

import com.stackanswer.retrofit.ListMovieResponse
import com.stackanswer.retrofit.ListShowResponse
import com.stackanswer.retrofit.detailmoviekt.DetailMovieResponse
import com.stackanswer.retrofit.detailshowkt.DetailShowResponse
import io.reactivex.Flowable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {

    @GET("3/movie/popular?")
    fun getListMoviePopuler(
            @Query("api_key") api_key: String?,
            @Query("language") language: String?,
            @Query("page") page: String?
    ): Flowable<ListMovieResponse?>?

    @GET("3/tv/popular?")
    fun getListShowPopuler(
            @Query("api_key") api_key: String?,
            @Query("language") language: String?,
            @Query("page") page: String?
    ): Flowable<ListShowResponse?>?

    @GET("3/tv/{id_movie}?")
    fun getDetailShowPopuler(
            @Path("id_movie") id_movie: String?,
            @Query("api_key") api_key: String?,
            @Query("language") language: String?
    ): Flowable<DetailShowResponse?>?

    @GET("3/movie/{id_movie}?")
    fun getDetailMoviePopuler(
            @Path("id_movie") id_movie: String?,
            @Query("api_key") api_key: String?,
            @Query("language") language: String?
    ): Flowable<DetailMovieResponse?>?
}
